lambda/publish-todays-curation/src/curation.ts

```diff
@@ -0,0 +1,132 @@
+import {CopyObjectCommand, HeadObjectCommand, NotFound, S3Client} from "@aws-sdk/client-s3";
+import {format, formatISO} from "date-fns";
+import { sendFastlyPurgeRequestWithRetries } from '@recipes-api/lib/recipes-data';
+import {Bucket, FastlyApiKey} from "./config";
+
+const s3Client = new S3Client({region: process.env["AWS_REGION"]});
+
+export interface CurationPath {
+  region: string;
+  variant: string;
+  year: number;
+  month: number;
+  day: number;
+}
+
+const KnownRegions = [
+  "northern",
+  "southern"
+];
+
+const KnownVariants = [
+  "meat-free",
+  "all-recipes"
+];
+
+const DateFormat = "yyyy-MM-dd";
+
+export async function validateAllCuration(date:Date, throwOnAbsent:boolean):Promise<CurationPath[]> {
+  const promises = KnownRegions.flatMap((region)=>
+    KnownVariants.map(async (variant) => {
+      const maybeInfo = await validateCurationData(region, variant, date);
+      if (!maybeInfo) {
+        console.warn(`No curation was present for region ${region} variant ${variant} on date ${format(date, DateFormat)}`);
+        if (throwOnAbsent) throw new Error(`Missing some curation for ${format(date, DateFormat)}. Consult the logs for more detail.`);
+      }
+      return maybeInfo;
+    }
+  ));
+
+  const allCurations = await Promise.all(promises);
+  return allCurations.filter((c)=>!!c) as CurationPath[];
+}
+
+export function generatePath(region:string, variant:string, date:Date) {
+  return `${region}/${variant}/${format(date, DateFormat)}/curation.json`;
+}
+
+function zeroPad(num:number, places:number) {
```

Comment:
reminds me of [this,](https://qz.com/646467/how-one-programmer-broke-the-internet-by-deleting-a-tiny-piece-of-code) he he he

Reply:
😆 that's why i write it myself!